test/e2e/multikueue/e2e_test.go

```diff
@@ -287,16 +289,36 @@ var _ = ginkgo.Describe("MultiKueue", func() {
 				}, util.Timeout, util.Interval).Should(gomega.Succeed())
 			})
 
+			ginkgo.By("Waiting for the jobSet to get status updates", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					createdJobset := &jobset.JobSet{}
+					g.Expect(k8sManagerClient.Get(ctx, client.ObjectKeyFromObject(jobSet), createdJobset)).To(gomega.Succeed())
+
+					g.Expect(createdJobset.Status.ReplicatedJobsStatus).To(gomega.BeComparableTo([]jobset.ReplicatedJobStatus{
```

Comment:
I'm wondering this could be rather hard to maintain so that it does not flake with kubernetes or JobSet changes. 

No strong view, but I think it is enough to check a stable subset of fields, just to prove there was an update. We could say use opts to limit the compare just to these fields (or even a subset of them).